VerifiableCredential/index.mjs

```diff
@@ -0,0 +1,277 @@
+import AJV from "ajv";
+import JSONPath from "jsonpath";
+import { LitElement, css, html, nothing } from "lit";
+import { ifDefined } from 'lit/directives/if-defined.js';
+import { map } from 'lit/directives/map.js';
+import { when } from "lit/directives/when.js";
+
+const VERSION = "https://identity.foundation/credential-manifest/spec/v1.0.0/";
+
+function isURL(string) {
+	try {
+		new URL(string);
+	} catch {
+		return false;
+	}
+	return true;
+}
+
+function verifyType(value, typeOrValidator) {
+	if (typeof typeOrValidator === "string" && typeof value !== typeOrValidator)
+		return undefined;
+	if (typeof typeOrValidator === "function" && !typeOrValidator(value))
+		return undefined;
+	return value;
+}
+
+function isHexColor(value) {
+	return /^#[0-9A-F]{6}$/i.test(value);
+}
+
+export class VerifiableCredential extends LitElement {
+	static properties = {
+		src: { type: String, attribute: "src" },
+		data: { type: Object, attribute: false },
+
+		manifest: { type: String, attribute: "manifest" },
+	};
+
+	static styles = css`
+.descriptor {
+	box-sizing: border-box;
+	display: flex;
+	flex-direction: column;
+	justify-content: flex-end;
+	position: relative;
+	max-width: min(100%, calc(var(--hero-width, 100%) * 1px));
+	max-height: min(100%, calc(var(--hero-height, 100%) * 1px));
+	aspect-ratio: var(--hero-width, 1) / var(--hero-height, 1);
+	padding: 1em;
+}
+
+.thumbnail {
+	position: absolute;
+	top: 1em;
+	left: 1em;
+	z-index: 2;
+	max-width: 32px;
+	max-height: 32px;
+}
+
+.hero {
+	position: absolute;
+	top: 0;
+	right: 0;
+	bottom: 0;
+	left: 0;
+	z-index: 1;
+	max-width: 100%;
+	max-height: 100%;
+}
+
+.title {
+	position: relative;
+	z-index: 3;
+	margin: 0;
+	padding: 0;
+	white-space: nowrap;
+	text-overflow: ellipsis;
+	overflow: hidden;
+}
+
+.subtitle {
+	position: relative;
+	z-index: 3;
+	margin: 0;
+	padding: 0;
+	white-space: nowrap;
+	text-overflow: ellipsis;
+	overflow: hidden;
+}
+
+.description {
+	position: relative;
+	z-index: 3;
+	margin: 0;
+	padding: 0.5em 0 0;
+	white-space: nowrap;
+	text-overflow: ellipsis;
+	overflow: hidden;
+}
+
+.properties {
+	position: relative;
+	z-index: 3;
+	margin: 0;
+	padding: 0.5em 0 0;
+}
+
+.property {
+	list-style-type: none;
+	white-space: nowrap;
+	text-overflow: ellipsis;
+	overflow: hidden;
+}
+
+.property::before {
+	margin: 0 0.3em 0 0.5em;
+	content: "\\2022";
+}
+`;
+
+	static shadowRootOptions = {
+		...LitElement.shadowRootOptions,
+		mode: "closed",
+	};
+
+	shouldUpdate(changedProperties) {
+		let shouldUpdate = true;
+
+		if (changedProperties.has("src")) {
+			shouldUpdate = false;
+
+			fetch(this.src)
+				.then((response) => response.json())
+				.then((json) => {
+					this.data = json;
+				});
```

Comment:
in both of those cases i believe the `response.json()` will throw, preventing `this.data` (and `this.manifest`) from being set(/overridden)

i was toying with the idea of having some sort of error display (e.g. "the data at the given `src` is not valid JSON") but felt that a console error is fine for now (especially since i imagine that developers would probably rather render nothing instead of some error text if something goes wrong)